lib/pbench/agent/tool_meister.py

```diff
@@ -427,49 +405,56 @@ def start(self):
             "-c",
             f"{self.pbench_install_dir}/templates/pmlogger.conf",
             "--host=localhost:55677",
-            f"{tool_dir}/%Y%m%d.%H.%M",
+            f"{_tool_dir}/%Y%m%d.%H.%M",
         ]
 
         self.logger.info(
             "%s: start_tool -- '%s' && '%s'",
             self.name,
-            " ".join(self.pmcd_args),
-            " ".join(self.pmlogger_args),
+            " ".join(pmcd_args),
+            " ".join(pmlogger_args),
         )
         self.pmcd_process = self._create_process_with_logger(
-            self.pmcd_args, tool_dir, "pmcd"
+            pmcd_args, _tool_dir, "pmcd"
         )
         self.pmlogger_process = self._create_process_with_logger(
-            self.pmlogger_args, tool_dir, "pmlogger"
+            pmlogger_args, _tool_dir, "pmlogger"
         )
 
     def stop(self):
         """Stop the pmcd and pmlogger processes.
         """
-        if self.pmcd_process is None:
-            raise ToolException(
-                f"Tool({self.name}) the expected pmcd process is not running"
-            )
-        if self.pmlogger_process is None:
-            raise ToolException(
-                f"Tool({self.name}) the expected pmlogger process is not running"
-            )
+        assert (
+            self.pmcd_process is not None
+        ), f"Tool({self.name}) the expected pmcd process is not running"
+        assert (
+            self.pmlogger_process is not None
+        ), f"Tool({self.name}) the expected pmlogger process is not running"
 
         self.logger.info("%s: stop_tool", self.name)
         try:
             self.pmlogger_process.terminate()
-        except Exception:
+        except (Warning, RuntimeError, OSError):
```

Comment:
Are we actually doing things that generate `Warning` exceptions? Should we fix them?

And while I appreciate the effort to use more specific classes than `Exception`, all three of these leave so much latitude that it's not really a massive improvement in documentation or clarity.

Reply:
It was a grand experiment for this change, but perhaps we could move this discussion to PR #2706?